src/core/Akka/Actor/ActorRef.cs

```diff
@@ -85,14 +94,23 @@ protected override void TellInternal(object message, ActorRef sender)
             {
                 if (Interlocked.Exchange(ref status, COMPLETED) == INITIATED)
                 {
-                    _unregister();
```

Comment:
I might have borked something here. 
Github PR compare for finding changes FTW...